File: 1-iniciante/1018.cpp
#include <iostream>
#include <vector>
#include<stdio.h>
using namespace std;
// 
int main()
{
    int cre;
    cin>>cre;
    cout<<cre<<endl;
    auto decompor=[cre]()->vector<int>{
        vector<int>deco={0,0,0,0,0,0,0}, var={100,50,20,10,5,2,1,0};
        int trem=cre;
        for(int i=0;i<7;i++){
            for(int j=i+1;j<8;j++){
                if(trem>=var[i]-var[j]){
                    deco[i]=1*int(trem/var[i]);
                    trem-=var[i]*int(trem/var[i]);
                    break;
                }
            }
            cout<<deco[i]<<" nota(s) de R$ "<<var[i]<<".00"<<endl;
        }
        return deco;
    };
    auto x=decompor();
    return 0;
}

File: 1-iniciante/1018.py
x=int(input())
v=[100,50,20,10,5,2,1,0]
deco=[0,0,0,0,0,0,0]
print(x)
for i in range(0,len(v)-1,1):
    for j in range(i+1,len(v),1):
        if x>=v[i]-v[j]:
            deco[i]=1*int(x/v[i])
            x-=v[i]*int(x/v[i])
    print(deco[i],"nota(s) de R$",v[i],end='')
    print(",00")